src/java/org/streamingpool/core/service/impl/TrackKeepingDiscoveryService.java

```diff
@@ -76,6 +70,7 @@ private TrackKeepingDiscoveryService(List<StreamFactory> factories, PoolContent
         this.idsOfStreamsUnderCreation = Collections.unmodifiableSet(idsOfStreamsUnderCreation);
         this.contextOfExecution = requireNonNull(contextOfExecution, "contextOfExecution must not be null");
         this.poolConfiguration = poolConfiguration;
+        this.idsDiscovered = new HashSet<>();
```

Comment:
I would tend to move this new HashSet() to the declaration line, I think it would be more clear that this will be new always....